### DISCUSSION TITLE
Can't get it to work

### DISCUSSION THREAD

**Discussion by Stevemoretz** [Laravel-Lang/common#14] [Q&A] [answered]
### Environment

- Laravel-Lang Version: 12.1.2
- Laravel Version: 8.83.24
- PHP Version: 7.4

### Issue description

I can't seem to find how to place the files to make it work, it used be a few php files when I used to use laravel 6 but now I have no idea where to put your files? 

```
When installing the package this way, you will need to manually copy the files from the vendor/laravel-lang/lang folder into your application.
```

The documentation doesn't mention it either, and it is vague.

Here are the structures I have tried:

<img width="280" alt="Screen Shot 1401-08-05 at 00 30 01" src="https://user-images.githubusercontent.com/27680142/198136755-262f5ba3-a6b6-4b44-943f-8e1ae6996d45.png">

### Steps to reproduce

I posted the screenshot in issue description.

**Comment by andrey-helldar** [comment]
Just use our installer: https://publisher.laravel-lang.com/installation/

```bash
composer require laravel-lang/publisher laravel-lang/lang laravel-lang/attributes --dev

php artisan lang:update
```

We wrote about it on the installation page: https://laravel-lang.com/installation/

![image](https://user-images.githubusercontent.com/10347617/198149129-ed2c9848-d30c-4200-ab43-7d0fde86e8e4.png)

**Reply by andrey-helldar** [reply]
What version of PHP and Laravel do you currently have?

https://publisher.laravel-lang.com/installation/compatibility.html

**Reply by Stevemoretz** [reply]
> What version of PHP and Laravel do you currently have?
> 
> https://publisher.laravel-lang.com/installation/compatibility.html

Thanks for the answer, my php version is 7.4 and Laravel is 8.x so I tried : 

```bash
composer require laravel-lang/lang:^10.2 laravel-lang/publisher:^12.0 --dev
```
Based on: https://laravel-lang.com/installation/compatibility.html#_10-0

And it got installed but https://publisher.laravel-lang.com/installation/ mentions I need to install laravel-lang/attributes so I found its docs https://attributes.laravel-lang.com/installation/compatibility.html running the command:

```bash
composer require laravel-lang/attributes:^2.0
```

gives 

```
./composer.json has been updated
Running composer update laravel-lang/attributes
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires laravel-lang/attributes ^2.0 -> satisfiable by laravel-lang/attributes[v2.0.0, ..., 2.x-dev].
    - laravel-lang/attributes[v2.0.0, ..., 2.x-dev] conflict with laravel-lang/publisher <14.0.0.
    - laravel-lang/publisher is locked to version v12.2.1 and an update of this package was not requested.


Installation failed, reverting ./composer.json and ./composer.lock to their original content.

```

**Reply by andrey-helldar** [reply]
Try to this:

```bash
composer require laravel-lang/publisher:^12.0 laravel-lang/lang:^10.2 laravel-lang/attributes:^1.0 --dev
```

**Reply by Stevemoretz** [reply]
> Try to this:
> 
> ```shell
> composer require laravel-lang/publisher:^12.0 laravel-lang/lang:^10.2 laravel-lang/attributes:^1.0 --dev
> ```

laravel-lang/attributes:^1.0 got installed thanks!

**Reply by andrey-helldar** [reply]
You're welcome!

**Comment by andrey-helldar** [comment, marked as answer]
PHP 8.1, 8.2:

```bash
composer require laravel-lang/publisher laravel-lang/lang laravel-lang/attributes --dev
```

PHP 8.0:

```bash
composer require laravel-lang/publisher:^13.0 laravel-lang/lang:^10.2 laravel-lang/attributes:^1.0 --dev
```

PHP 7.4:

```bash
composer require laravel-lang/publisher:^12.0 laravel-lang/lang:^10.2 laravel-lang/attributes:^1.0 --dev
```

**Comment by nxmndr** [comment]
Just a heads up that from the outside this has gotten quite confusing ; the information is spread out etc. I can't find a quickstart page where all of it is available, although it fits in 3 lines :

```
composer require laravel-lang/publisher laravel-lang/attributes laravel-lang/lang --dev
./artisan lang:add fr
git add composer.json resources/lang/
```

**Reply by andrey-helldar** [reply]
1. https://laravel-lang.com/installation/
2. 
![image](https://user-images.githubusercontent.com/10347617/210590100-d06d3a95-a8da-40f0-8ec7-e1300a4eaf67.png)
3. https://publisher.laravel-lang.com/installation/

**Reply by nxmndr** [reply]
On 3 different pages, with 3 new packages to check out. I guarantee you newbies will spend more time they need on this. A 3-lines quickstart will cover it for most people. Should be on laravel-lang/lang README too.

**Reply by andrey-helldar** [reply]
On the one hand, you are right, but on the other hand, the Lang project has nothing to do with either Attributes or Publisher. Just like the Attributes project has nothing to do with either Lang or Publisher.

But the Publisher project can set keys from projects, including from Lang and Attributes.

As for beginners, it's enough to read :)

![image](https://user-images.githubusercontent.com/10347617/210595506-a6a8f814-e73f-445a-b118-88f2cf3a23f9.png)
